### ISSUE TITLE
[BUG] flutter 3.0 analyzer plugin does not seem to work

### ISSUE THREAD

**Issue by fzyzcjy** [dart-code-checker/dart-code-metrics#846]
* **Dart code metrics version:** 4.15.0
* **Dart sdk version:** flutter3.0

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```yaml
analyzer:
  plugins:
    - dart_code_metrics
...
```

</details>

**What did you do? Please include the source code example causing the issue.**

```dart

```

**What did you expect to happen?**

**What actually happened?**

**Are you willing to submit a pull request to fix this bug?**

Hi thanks for the lib! However today it seems that the analyzer plugin does not work. Even if I write code that should lint, it does not.

**Comment by fzyzcjy** [comment]
Here is what analyzer server says:

```
Plugins
Plugins in use.

Analysis plugins
dart_code_metrics-4.15.0
bootstrap package path: /Users/tom/.pub-cache/hosted/pub.flutter-io.cn/dart_code_metrics-4.15.0/tools/analyzer_plugin

execution path:

packages file path:

not running

ProcessException: No such file or directory
  Command: /Users/tom/fvm/default/bin/cache/dart-sdk/bin/pub upgrade
#0      _ProcessImpl._runAndWait (dart:io-patch/process_patch.dart:487:7)
#1      _runNonInteractiveProcessSync (dart:io-patch/process_patch.dart:632:18)
#2      Process.runSync (dart:io-patch/process_patch.dart:68:12)
#3      PluginManager._computeFiles (package:analysis_server/src/plugin/plugin_manager.dart:641:28)
#4      PluginManager.filesFor (package:analysis_server/src/plugin/plugin_manager.dart:462:14)
#5      PluginManager.addPluginToContextRoot (package:analysis_server/src/plugin/plugin_manager.dart:328:23)
#6      PluginWatcher.addedDriver (package:analysis_server/src/plugin/plugin_watcher.dart:68:19)
#7      AnalysisDriver.configure (package:analyzer/src/dart/analysis/driver.dart:555:33)
#8      ContextBuilderImpl.createContext (package:analyzer/src/dart/analysis/context_builder.dart:137:12)
#9      new AnalysisContextCollectionImpl (package:analyzer/src/dart/analysis/analysis_context_collection.dart:67:36)
#10     ContextManagerImpl._createAnalysisContexts.performContextRebuildGuarded.performContextRebuild (package:analysis_server/src/context_manager.dart:449:40)
#11     ContextManagerImpl._createAnalysisContexts.performContextRebuildGuarded.performContextRebuild (package:analysis_server/src/context_manager.dart:444:41)
#12     ContextManagerImpl._createAnalysisContexts.performContextRebuildGuarded (package:analysis_server/src/context_manager.dart:562:17)

#13     ContextManagerImpl.setRoots (package:analysis_server/src/context_manager.dart:280:5)

#14     AnalysisServer.setAnalysisRoots (package:analysis_server/src/analysis_server.dart:456:7)

#15     AnalysisSetAnalysisRootsHandler.handle (package:analysis_server/src/handler/legacy/analysis_set_analysis_roots.dart:50:7)


dart_code_metrics-4.15.1
bootstrap package path: /Users/tom/.pub-cache/hosted/pub.flutter-io.cn/dart_code_metrics-4.15.1/tools/analyzer_plugin

execution path:

packages file path:

not running

ProcessException: No such file or directory
  Command: /Users/tom/fvm/default/bin/cache/dart-sdk/bin/pub upgrade
#0      _ProcessImpl._runAndWait (dart:io-patch/process_patch.dart:487:7)
#1      _runNonInteractiveProcessSync (dart:io-patch/process_patch.dart:632:18)
#2      Process.runSync (dart:io-patch/process_patch.dart:68:12)
#3      PluginManager._computeFiles (package:analysis_server/src/plugin/plugin_manager.dart:641:28)
#4      PluginManager.filesFor (package:analysis_server/src/plugin/plugin_manager.dart:462:14)
#5      PluginManager.addPluginToContextRoot (package:analysis_server/src/plugin/plugin_manager.dart:328:23)
#6      PluginWatcher.addedDriver (package:analysis_server/src/plugin/plugin_watcher.dart:68:19)
#7      AnalysisDriver.configure (package:analyzer/src/dart/analysis/driver.dart:555:33)
#8      ContextBuilderImpl.createContext (package:analyzer/src/dart/analysis/context_builder.dart:137:12)
#9      new AnalysisContextCollectionImpl (package:analyzer/src/dart/analysis/analysis_context_collection.dart:67:36)
#10     ContextManagerImpl._createAnalysisContexts.performContextRebuildGuarded.performContextRebuild (package:analysis_server/src/context_manager.dart:449:40)
#11     ContextManagerImpl._createAnalysisContexts.performContextRebuildGuarded.performContextRebuild (package:analysis_server/src/context_manager.dart:444:41)
#12     ContextManagerImpl._createAnalysisContexts.performContextRebuildGuarded (package:analysis_server/src/context_manager.dart:562:17)

#13     ContextManagerImpl.setRoots (package:analysis_server/src/context_manager.dart:280:5)

#14     AnalysisServer.setAnalysisRoots (package:analysis_server/src/analysis_server.dart:456:7)

#15     AnalysisSetAnalysisRootsHandler.handle (package:analysis_server/src/handler/legacy/analysis_set_analysis_roots.dart:50:7)

**Comment by fzyzcjy** [comment]
Analysis server version: 1.33.0
Dart SDK: 2.17.0 (stable) (Mon May 9 10:36:47 2022 +0200) on "macos_x64"

**Comment by fzyzcjy** [comment]
Duplicate of https://github.com/dart-lang/sdk/issues/48682

**Comment by incendial** [comment]
> Duplicate of https://github.com/dart-lang/sdk/issues/48682

Happy to see that this is not a new problem 😅

**Comment by fzyzcjy** [comment]
😅
